src/daft-shuffles/src/shuffle_cache.rs

```diff
@@ -43,19 +67,23 @@ struct InProgressShuffleCacheState {
 
 pub struct InProgressShuffleCache {
     state: Mutex<InProgressShuffleCacheState>,
+    shuffle_dirs: Vec<String>,
     partitioner_sender_weak: async_channel::WeakSender<Arc<MicroPartition>>,
 }
 
 impl InProgressShuffleCache {
     pub fn try_new(
         num_partitions: usize,
         dirs: &[String],
+        node_id: String,
+        shuffle_stage_id: usize,
         target_filesize: usize,
         compression: Option<&str>,
         partition_by: Option<Vec<ExprRef>>,
     ) -> DaftResult<Self> {
         // Create the directories
-        for dir in dirs {
+        let shuffle_dirs = get_shuffle_dirs(dirs, &node_id, shuffle_stage_id);
+        for dir in &shuffle_dirs {
```

Comment:
let's also make a note that if a directory fills up (maybe periodically check the dirs) , we should favor the other ones. And if all of the them are near full, we should raise an error (rather than the Out of disk killer who will come and get us)